store/neo4j.go

```diff
@@ -0,0 +1,201 @@
+package store
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/ONSdigital/dp-code-list-api/datastore"
+	"github.com/ONSdigital/dp-code-list-api/models"
+	"github.com/ONSdigital/go-ns/log"
+	bolt "github.com/johnnadratowski/golang-neo4j-bolt-driver"
+	"github.com/johnnadratowski/golang-neo4j-bolt-driver/structures/graph"
+	"github.com/pkg/errors"
+)
+
+const (
+	getCodeListsQuery = "MATCH (i) WHERE i:_%s%s RETURN distinct labels(i) as labels, i"
+	getCodeListQuery  = "MATCH (i:_%s:`_name_%s`) RETURN i"
+)
+
+// NeoDataStore represents the necessary information to access
+// neo4j
+type NeoDataStore struct {
+	pool          DBPool
+	codeListLabel string
+}
+
+// DBPool contains the methods to control access to the Neo4J
+// database pool
+type DBPool interface {
+	OpenPool() (bolt.Conn, error)
+	Close() error
+}
+
+// Close is a wrapper for the neo pool close
+func (n *NeoDataStore) Close() error {
+	return n.pool.Close()
+}
+
+// CreateNeoDataStore allows the creation of a NeoDataStore
+func CreateNeoDataStore(addr, codelistLabel string, conns int) (n *NeoDataStore, err error) {
+	store, err := bolt.NewClosableDriverPool(addr, conns)
+	if err != nil {
+		return
+	}
+
+	n = &NeoDataStore{
+		pool:          store,
+		codeListLabel: codelistLabel,
+	}
+
+	return
+}
+
+// GetCodeLists returns a list of code lists
+func (n *NeoDataStore) GetCodeLists(ctx context.Context, filterBy string) (*models.CodeListResults, error) {
+	logData := log.Data{}
+	if len(filterBy) > 0 {
+		logData = log.Data{"filter_by": filterBy}
```

Comment:
We usually seem to say `logData["filter_by"] = ` rather than the struct declaration. Is there a reason to do this differently here?